CMakeLists.txt

```diff
@@ -35,14 +35,22 @@ include(cmake/setup_build.cmake)
 include(cmake/utils.cmake)
 include(CheckCXXCompilerFlag)
 
-# This flag is not supported on Windows and some older version of GCC
+# These flags are not supported on Windows and some older version of GCC
 # that our bots use.
+# Warning about implicit fallthrough in switch blocks
 unset(COMPILER_SUPPORTS_FALLTHROUGH_WARNING CACHE)
 check_cxx_compiler_flag(-Wimplicit-fallthrough COMPILER_SUPPORTS_FALLTHROUGH_WARNING)
 if (COMPILER_SUPPORTS_FALLTHROUGH_WARNING)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wimplicit-fallthrough")
 endif()
 
+# Warning about extra semi-colons
+unset(COMPILER_SUPPORTS_EXTRA_SEMI_WARNING CACHE)
```

Comment:
Slight concern: Does this Cmake variable name need to be scoped to Shaderc?  
I guess there's no extra harm in that there is an existing usage for the fall-through warning. So I'll let this through.

Reply:
I think it gets scoped by default, maybe? I don't fully grok how CACHE and scoping interact.

I am honestly not sure why these variables are CACHE, since they are getting unset unconditionally every time. I am going to file a bug to investigate this, since this might be something that can be improved.